sdk/core/azure-core/inc/azure/core/md5.hpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Utility functions to help compute the MD5 hash value for the input binary data.
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <vector>
+
+namespace Azure { namespace Core {
+
+  /**
+   * @brief Represents the class for the MD5 hash function which maps binary data of an arbitrary
+   * length to small binary data of a fixed length.
+   */
+  class Md5 {
+  public:
+    /**
+     * @brief Construct a default instance of @Md5.
+     */
+    Md5();
+
+    /**
+     * @brief Cleanup any state when destroying the instance of @Md5.
+     */
+    ~Md5();
+
+    /**
+     * @brief Used to append parital binary input data to compute the hash in a streaming fashion.
+     * @param data The input vector that contains the current block of binary data that is used for
+     * hash calculation.
+     * @remark Once all the data has been added, call #Digest() to get the computed hash value.
+     */
+    void Update(const std::vector<uint8_t>& data);
+
+    /**
+     * @brief Computes the hash value of all the binary input data appended to the instance so far.
+     * @return The computed MD5 hash value corresponding to the input provided.
+     * @remark Use #Update() to add more partial data before calling this function.
```

Comment:
Make sure that `#Udpate()` works without the full path to it like `#Azure::Core::Md5::Update`

Reply:
Do we have documented steps on how to generate docs locally, that you can share?

Reply:
Just enable the BUILD_DOCUMENTATION option for generating and make sure to have doxygen ver 1.8.18 at least

I don't think we have other but the option mentioned in the docs.  I can add some docs in the future if you create an issue :)
Otherwise, we can chat and I can help

Reply:
> Make sure that `#Udpate()` works without the full path to it like `#Azure::Core::Md5::Update`

It works fine:
![image](https://user-images.githubusercontent.com/6527137/107422914-8ad6b200-6ad0-11eb-8e7e-b3563be3cd69.png)